source/server/admin/stats_handler.cc

```diff
@@ -187,8 +198,20 @@ void StatsHandler::statsAsText(const std::map<std::string, uint64_t>& all_stats,
   std::map<std::string, std::string> all_histograms;
   for (const Stats::ParentHistogramSharedPtr& histogram : histograms) {
     if (shouldShowMetric(*histogram, used_only, regex)) {
-      auto insert = all_histograms.emplace(histogram->name(), histogram->quantileSummary());
-      ASSERT(insert.second); // No duplicates expected.
+      bool success = false;
+      // Display bucket data if histogram_buckets query parameter is used, otherwise output contains
+      // quantile summary data.
+      if (!histogram_buckets_value.has_value()) {
+        success = all_histograms.emplace(histogram->name(), histogram->quantileSummary()).second;
+      } else if (histogram_buckets_value.value() == "cumulative") {
+        success = all_histograms.emplace(histogram->name(), histogram->bucketSummary()).second;
+      } else {
+        ASSERT(histogram_buckets_value.value() ==
```

Comment:
It doesn't seem great that a typo in the URL bar crashes a server compiled for debug. Can you just make this ENVOY_LOG and return a 400 with a terse error?

Reply:
I added a clarifying comment near the ASSERT as I implemented the checking elsewhere:

https://github.com/envoyproxy/envoy/blob/7401db0bb46a34f81918a745d6378bd1f89bc6a1/source/server/admin/stats_handler.cc#L88-L94

If the value is "disjoint" at this point something has gone wrong. Is this fine?